storage/tool/src/main/java/com/hortonworks/streamline/storage/tool/SchemaMigrationHelper.java

```diff
@@ -0,0 +1,111 @@
+package com.hortonworks.streamline.storage.tool;
+
+import com.hortonworks.streamline.storage.tool.exception.SchemaMigrationException;
+import org.flywaydb.core.Flyway;
+import org.flywaydb.core.internal.dbsupport.FlywaySqlException;
+
+import static org.flywaydb.core.internal.info.MigrationInfoDumper.dumpToAsciiTable;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class SchemaMigrationHelper {
+
+    private Flyway flyway;
+
+    public SchemaMigrationHelper(Flyway flyway) {
+        this.flyway = flyway;
+    }
+
+    private void create() throws SQLException {
+        if(!validateDatabase(true))
+            throw new SchemaMigrationException("Target database is not empty, please consider executing with \"migrate\" option");
+        flyway.migrate();
+    }
+
+    private void migrate() throws SQLException {
+        if(!validateDatabase(false))
+            throw new SchemaMigrationException("Target database is empty, please consider executing with \"create\" option");
+        flyway.migrate();
+    }
+
+    private void clean() {
+        flyway.clean();
+    }
+
+    private boolean checkConnection() throws SQLException {
+        try (Connection connection = flyway.getDataSource().getConnection()) {
+            return true;
+        } catch (FlywaySqlException e) {
+            System.out.println("Connection failure: " + e.getMessage());
+            return false;
+        }
+    }
+
+    private void info() {
+        System.out.println(dumpToAsciiTable(flyway.info().all()));
+    }
+
+    private void validate() {
+        flyway.validate();
+    }
+
+    private void repair() {
+        flyway.repair();
+    }
+
+    private boolean isDatabaseEmpty(Connection connection) throws SQLException {
+        DatabaseMetaData databaseMetaData = connection.getMetaData();
+        try (ResultSet resultSet = databaseMetaData.getTables(connection.getCatalog(), connection.getSchema(), "", null)) {
+            if (resultSet.next())
+                return false;
+            return true;
+        } catch (SQLException e) {
+            throw new SchemaMigrationException("Unable the obtain the state of the target database", e);
+        }
+    }
+
+    private boolean validateDatabase(Boolean checkDatabaseEmpty) throws SQLException {
```

Comment:
I dont see any validations happening here. If this is to have a single method to check if database is empty or not I think we can just have a `isDatabaseEmpty()` that returns true/false.